backend/src/server.cpp

```diff
@@ -42,12 +42,25 @@ namespace server
             return response;
         }
 
-        std::string encrypted_password = cryptography::encryptPassword(crow::utility::base64decode(user_password, user_password.size()), b);
-        // std::string encrypted_password = cryptography::encryptPassword("TestPass1&", b);
+        // get all passwords from database
+        std::function <std::string(sqlite3_stmt *)> callback = [](sqlite3_stmt *stmt) {
+            return std::string(reinterpret_cast<const char *>(sqlite3_column_text(stmt, 0)));
+        };
+        std::vector<std::string> result = db.execute("SELECT * FROM passwords;", callback);
+
+        // get b secret key from database
+        std::string encodedB = db.execute("SELECT * FROM secret;", callback)[0];
+        // decode secret key b
+        std::string decodedB = crow::utility::base64decode(encodedB, encodedB.size());
 
+        // copy secret key b into an unsigned char* 
+        unsigned char *b = (unsigned char *)decodedB.data();
```

Comment:
**thought**: We need to get rid of all these C-style parenthetical casts at some point.